portal-impl/test/unit/com/liferay/portal/util/PortalImplUnitTest.java

```diff
@@ -147,11 +147,14 @@ public void testGetForwardedHostWithCustomXForwardedHostEnabled()
 			PropsValues.WEB_SERVER_FORWARDED_HOST_ENABLED;
 		String webServerForwardedHostHeader =
 			PropsValues.WEB_SERVER_FORWARDED_HOST_HEADER;
+		String[] virtualHostsValidHosts = PropsValues.VIRTUAL_HOSTS_VALID_HOSTS;
 
 		try {
 			setPropsValuesValue("WEB_SERVER_FORWARDED_HOST_ENABLED", true);
 			setPropsValuesValue(
 				"WEB_SERVER_FORWARDED_HOST_HEADER", "X-Forwarded-Custom-Host");
+			setPropsValuesValue(
```

Comment:
The reason we do not use mock here is, when we call _portalImpl.getForwardedHost():

**Before upgrading virtual.hosts.vali.hosts**
forwardedServer is considered as VIRTUAL_HOSTS_VALID_HOSTS, because the default value of it is *. Also, `_validPortalDomainCheckDisabled` is set to be true, which means it will just `return forwardedHost` after going in to `if (_validPortalDomainCheckDisabled)` in `_portalImpl.getForwardedHost()`

**After upgrading virtual.hosts.vali.hosts**
forwardedServer is **not** part of valid host by default, and `_validPortalDomainCheckDisabled` is set to be false, which means it will go into `else if (!isValidPortalDomain(forwardedHost))`, and return false at the end. 

The consistent test logic as before, should set forwardedServer as validhost, so that it will return true while checking if it is valid, and it won't go into 
```
if (StringUtil.equalsIgnoreCase(domain, PropsValues.WEB_SERVER_HOST) ||
			isValidVirtualHostname(domain) ||
			StringUtil.equalsIgnoreCase(domain, getCDNHostHttp(companyId)) ||
			StringUtil.equalsIgnoreCase(domain, getCDNHostHttps(companyId)))
```
and will not cause NPE.